### PR TITLE
Switch to compile time evaluated filename stem

### PR DESCRIPTION
**Pull request by rageagainsthepc**

### CODE REVIEW

File: vmicore/src/io/ILogger.h [hunk 1]
```diff
@@ -4,63 +4,63 @@
 #include "cxxbridge/rust/cxx.h"
 #include "cxxbridge/rust_grpc_server/src/bridge.rs.h"
 #include <memory>
+#include <string_view>
 
 namespace VmiCore
 {
-#define NEW_LOGGER(log_server) ((log_server)->newNamedLogger(std::filesystem::path(__FILE__).filename().stem()))
     // Has to be the same as in the HiveOperations project to indicate in which file the log should be written.
```
Comment (K-Mayer) [inline vmicore/src/io/ILogger.h:11]: I think we should try and get rid of these processing internals.

File: vmicore/src/io/grpc/GRPCServer.cpp [hunk 2]
```diff
@@ -70,13 +71,13 @@ namespace VmiCore
...
 
-    void GRPCServer::sendErrorEvent(const std::string& message)
+    void GRPCServer::sendErrorEvent(const std::string_view& message)
     {
-        (*server)->send_error_event(message);
+        (*server)->send_error_event(static_cast<std::string>(message));
     }
 
-    void GRPCServer::sendInMemDetectionEvent(const std::string& message)
+    void GRPCServer::sendInMemDetectionEvent(const std::string_view& message)
```
Comment (K-Mayer) [inline vmicore/src/io/grpc/GRPCServer.cpp:79]: Is there a specific reason you did not update the plugin-system/the plugin to the string_view in this PR as well?

File: vmicore/test/io/mock_Logging.h [hunk 3]
```diff
@@ -13,7 +13,7 @@ namespace VmiCore
         MOCK_METHOD(void, start, (), (override));
         MOCK_METHOD(void, stop, (const uint64_t&), (override));
         MOCK_METHOD(std::unique_ptr<ILogger>, newLogger, (), (override));
-        MOCK_METHOD(std::unique_ptr<ILogger>, newNamedLogger, (const std::string&), (override));
+        MOCK_METHOD(std::unique_ptr<ILogger>, newNamedLogger, (const std::string_view&), (override));
```
Comment (K-Mayer) [inline vmicore/test/io/mock_Logging.h:16]: nit: inconsistent formatting (missing empty lines)

**Review by K-Mayer — APPROVED** [review]

File: vmicore/src/io/ILogger.h (hunk 1 above)
Comment (rageagainsthepc) [reply vmicore/src/io/ILogger.h:11]: Agreed, but not in this PR though.

File: vmicore/src/io/grpc/GRPCServer.cpp (hunk 2 above)
Comment (rageagainsthepc) [reply vmicore/src/io/grpc/GRPCServer.cpp:79]: No

**Pull request merged**